updating to use Tracker parameters from DB

jshlee opened the pull request:
update of GeometryExtended2023Muon_cff.py left out from PR #7966

fireworks - updating geo dumpers both dumpSimGeometry and dumpRecoGeometry

cmsbuild commented:
A new Pull Request was created by @jshlee (Jason Lee) for CMSSW_7_5_X.

updating to use Tracker parameters from DB

It involves the following packages:

Configuration/Geometry


@civanch, @Dr15Jones, @ianna, @mdhildreth, @cmsbuild, @nclopezo can you please review it and eventually sign? Thanks.
@ghellwig, @Martin-Grunewald this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@nclopezo you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

cmsbuild commented:
Pull request #9236 was updated. @civanch, @Dr15Jones, @ianna, @mdhildreth, @cmsbuild, @alja, @nclopezo can you please check and sign again.

cmsbuild commented:
+1
Tested at: feab7ee3e8ba60f52e8c412549213305d61c2e61
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9236/4989/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9236/4989/summary.html

ianna commented on Fireworks/Geometry/python/dumpRecoGeometry_cfg.py:
```diff
@@ -35,18 +35,13 @@ def recoGeoLoad(score):
-      from Configuration.AlCa.GlobalTag import GlobalTag
-      process.GlobalTag = GlobalTag(process.GlobalTag, 'auto:upgrade2019', '')
+      from Configuration.AlCa.autoCond import autoCond
+      process.GlobalTag.globaltag = autoCond['mc']
```
@jshlee - this will pickup run1 geometry and not 2019. Please, discard this change.

ianna commented on Fireworks/Geometry/python/dumpRecoGeometry_cfg.py:
```diff
@@ -35,18 +35,13 @@ def recoGeoLoad(score):
     elif score == "2017":
        from Configuration.AlCa.GlobalTag import GlobalTag
-       process.GlobalTag = GlobalTag(process.GlobalTag, 'auto:upgrade2017', '')
+       process.GlobalTag = GlobalTag(process.GlobalTag, 'auto:mc', '')
```
@jshlee - this will pickup run1 geometry and not 2017. Please, discard this change.

alja commented:
-1

Need to fix dumpRecoGeometry_cfg.py  first.

jshlee commented:
Hi @ianna, 

Using auto:upgrade2017 and auto:upgrade2019 results in the following error
No "PTrackerParametersRcd" record found in the dependent record "TrackerDigiGeometryRecord".

Cheers,
Jason.

jshlee committed:
fixed globaltag for 2017,2019 and 2023
changed to autoCond['run2_design']
added PTrackerParametersRcd from frontier://FrontierProd/CMS_CONDITIONS

mmusich commented:
@jshlee this hack is currently needed just because we don't have yet a proper Global tag to use for the upgrade scenarios. Overriding the tags via `toGet` is ugly and also error prone (because at some point someone will loose track of why that particular tag was included). Is this PR urgent? we are working to get the upgrade autoCond keys to work in 75X.

cmsbuild commented:
Pull request #9236 was updated. @civanch, @Dr15Jones, @ianna, @mdhildreth, @cmsbuild, @alja can you please check and sign again.

jshlee commented:
@mmusich I agree it's ugly.
I'm don't think its urgent.
I guess @ianna or @pietverwilligen would be a better judge of this.

alja commented:
+1

cmsbuild commented:
+1
Tested at: f2e7ca5f2e4191cc586a4333ff23dfbe7736f60f
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9236/115/summary.html

ianna commented:
@jshlee - I'd suggest to remove `Fireworks/Geometry/python/dumpRecoGeometry_cfg.py` from this PR. It wouldn't be good to have this temporary fix in a release. Besides, it needs further fixes - auto[mc] GT is correct only for Run 1.
I'd suggest, while a new GT is in preparation, you keep your own branch for the moment and advertise it for all interested parties.

ianna commented:
-1

jshlee committed:
revert changes to dumpRecoGeometry_cfg.py

cmsbuild commented:
Pull request #9236 was updated. @civanch, @Dr15Jones, @ianna, @mdhildreth, @cmsbuild, @alja can you please check and sign again.

jshlee commented:
@ianna - done.

ianna commented:
+1

cmsbuild commented:
+1
Tested at: fb93d77e2673e00049ad7564616ca2b35d2cafff
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9236/2581/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9236/2581/summary.html
